src/main/java/pl/shop/models/CustomerDetails.java

```diff
@@ -0,0 +1,33 @@
+package pl.shop.models;
+
+import lombok.Data;
+
+import javax.persistence.*;
+import java.util.UUID;
+
+@Entity
+@Data
+public class CustomerDetails {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private UUID uuid;
+
+    @OneToOne
+    @MapsId
+    private Customer customer;
+
```

Comment:
Tworzę właśnie encję Order i widzę mały problem. W encji CustomerDetails są tylko dane adresowe i kontaktowe klienta, poza tym nic więcej.  Powinniśmy te dane wciągnąć do nowej encji o nazwie Address, dlatego, że gdy klient zażyczy sobie zamówienie na inny adres, to w aktualnym stanie nie możemy tego zrobić (relacja 1 do 1), chyba, że zmienimy dane adresowe klienta.  Tu w przyszłości jeszcze pewnie dojdzie pole Payment( w którym np będą przechowywane dane karty klienta ) oraz różne inne dodatkowe dane. Ja tą encję widzę tak CustomerDetails(UUid id, Address address, Payment payment ...)

Reply:
jasne, pewnie, w 100% się zgadzam. Na ten moment te encje posklejałem z umla, żeby było na czym pracować i żeby pokazać pare hibernate-owych adnotacji ale jak najbardziej wszystko podlega refaktoryzacji =]

pytanie do czego nam jest productsBought. Jeżeli to ma być historia zamówień to faktycznie, sam Set<Order> wystarczy. Jeżeli ma być do użytku wewnętrznego (np, wysyłanie powiadomień gdy coś co ktoś kupił jest na promocji, albo jako baza danych dla funkcji 'użytkownicy jak Ty kupili też to i tamto') to chyba tak jak jest teraz jest ok